scripts/e2e.mjs

```diff
@@ -0,0 +1,219 @@
+const appName = 'RnDiffApp';
+const appDir = `${appRepoDir}/${appName}`;
+const e2eDir = `${rootDir}/test/e2e`;
+const testAppName = `${appName}.${platform == 'ios' ? 'app' : 'apk'}`;
+
+// Build and publish the SDK - we only need to do this once in CI.
+// Locally, we may want to get updates from the latest build so do it on every app build.
+if (actions.includes('create') || (env.CI === undefined && actions.includes('build'))) {
+  execSync(`yarn build`, { stdio: 'inherit', cwd: rootDir, env: env });
+  execSync(`yalc publish`, { stdio: 'inherit', cwd: rootDir, env: env });
+  execSync(`yarn build`, { stdio: 'inherit', cwd: e2eDir, env: env });
+}
+
+if (actions.includes('create')) {
+  // Clone the test app repo
+  if (fs.existsSync(appRepoDir)) fs.rmSync(appRepoDir, { recursive: true });
+  execSync(`git clone ${appSourceRepo}  --branch release/${RNVersion} --single-branch ${appRepoDir}`, { stdio: 'inherit', env: env });
+
+  // Install dependencies
+  // yalc add doesn't fail if the package is not found - it skips silently.
+  const yalcAddOutput = execSync(`yalc add @sentry/react-native`, { cwd: appDir, env: env, encoding: 'utf-8' });
+  if (!yalcAddOutput.match(/Package .* added ==>/)) {
+    console.error(yalcAddOutput);
+    process.exit(1);
+  } else {
+    console.log(yalcAddOutput.trim());
+  }
+
+  // original yarnrc contains the exact yarn version which causes corepack to fail to install yarn v3
+  fs.writeFileSync(`${appDir}/.yarnrc.yml`, 'nodeLinker: node-modules', { encoding: 'utf-8' });
+  // yarn v3 won't install dependencies in a sub project without a yarn.lock file present
+  fs.writeFileSync(`${appDir}/yarn.lock`, '');
+
+  execSync(`yarn install`, {
+    stdio: 'inherit', cwd: appDir,
+    // yarn v3 run immutable install by default in CI
+    env: Object.assign(env, { YARN_ENABLE_IMMUTABLE_INSTALLS: false })
+  });
+
+  execSync(`yarn add ../../../../e2e`, { stdio: 'inherit', cwd: appDir, env: env });
+
+  // Patch the app
+  execSync(`patch --verbose --strip=0 --force --ignore-whitespace --fuzz 4 < ../../../rn.patch`, { stdio: 'inherit', cwd: appDir, env: env });
+  execSync(`../../../rn.patch.app.js --app .`, { stdio: 'inherit', cwd: appDir, env: env });
+  execSync(`../../../rn.patch.metro.config.js --path metro.config.js`, { stdio: 'inherit', cwd: appDir, env: env });
+
+  // Set up platform-specific app configuration
+  if (platform == 'ios') {
+    execSync('ruby --version', { stdio: 'inherit', cwd: `${appDir}`, env: env });
+
+    // Fixes Hermes pod install https://github.com/CocoaPods/CocoaPods/issues/12226#issuecomment-1930604302
+    execSync(`gem install cocoapods -v 1.15.2`, { stdio: 'inherit', cwd: appDir, env: env });
+    execSync(`../../../../rn.patch.podfile.js --pod-file Podfile --engine ${RNEngine}`, { stdio: 'inherit', cwd: `${appDir}/ios`, env: env });
+
+    // const podInstallCommand = RNVersion === '0.65.3' ? 'pod install' : 'bundle exec pod install';
```

Comment:
Since we are forcing specific cocoapods version using `gem install cocoapods...`the direct `pod install` will work for all versions (at the moment, when we change the pods version we will need to reevaluate)